lib/redis/store/factory.rb

```diff
@@ -78,13 +78,16 @@ def self.extract_host_options_from_uri(uri)
             :scheme   => uri.scheme,
             :host     => uri.hostname,
             :port     => uri.port || DEFAULT_PORT,
-            :password => uri.password.nil? ? nil : CGI.unescape(uri.password.to_s),
             :ssl      => uri.scheme == 'rediss'
           }
 
           options[:db]        = db.to_i   if db
           options[:namespace] = namespace if namespace
         end
+
+        options[:username] = uri.user if uri.user
+        options[:password] = CGI.unescape(uri.password.to_s) if uri.password
```

Comment:
Are username & password are supported for unix URLs?
https://github.com/redis/redis-rb/blob/v5.2.0/lib/redis.rb#L43-L44

Reply:
Did not see your comment above, collapsed by default